packages/material-ui/src/Typography/Typography.d.ts

```diff
@@ -22,6 +22,7 @@ export interface TypographyTypeMap<P = {}, D extends React.ElementType = 'span'>
     paragraph?: boolean;
     variant?: Variant | 'inherit';
     variantMapping?: Partial<Record<Variant, string>>;
+    component?: React.ElementType;
```

Comment:
This may defeat the whole purpose of these types. I think Typography isn't that well tested in that regard. `ButtonBase` was the first component where we implemented the TypeMap pattern. It might be worthwile exploring this idea for `Button` and `ButtonProps` if there are any.

Reply:
@eps1lon not clear on whether this is a request or just a comment and what to do about it. Can I help move this forward? Is there a solution to the breaking change that would get accepted and, can further work/improvements go to a separate detailed ticket?

Reply:
This was more talking to myself to take notes :laughing: I think I have some cases with undesired effects. I'll add these to this PR. I don't think we need perfectly sound types so this approach is probably fine. 

But we might be able to solve this for all affected components. This `SomeComponentProps: PropsTypeOfSomeComponent` is pretty common in our codebase so I'd rather make it right.